pkgs/development/python-modules/py-air-control/default.nix

```diff
@@ -0,0 +1,25 @@
+{ buildPythonPackage, coapthon3, fetchPypi, isPy27, lib, pycryptodomex }:
+
+buildPythonPackage rec {
+  pname = "py-air-control";
+  version = "2.1.0";
+  disabled = isPy27;
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0g9qkaygjyyxkhgx51m8hhj9inlzmynzw6l4vp7g1cflrn8g3i92";
+  };
+
+  propagatedBuildInputs = [ pycryptodomex coapthon3 ];
+
+  # Disabling tests because they are not included in the package.
+  doCheck = false;
```

Comment:
Please see if the github repo has tests

There are no tests included. Please try to checkout from source and check if they have unit tests, and try to run them. Unit tests give a good indication that they package has a high degree of validity and correctness given the python package set.

If tests are not available, then please use `pythonImportsCheck` to import the most important modules. This isn't as good as unit tests, but will usually give a good indication of run-time errors. Please see [pythonImportsCheck documentation](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pythonimportscheck) for more information.

Reply:
Oh, I was wondering about that. Yes, there are tests in Github (just not distributed in PyPi). I thought that we should always use `fetchPypi`, so my goal was to contribute to `py-air-control` and `coapthon3`, and make sure tests are included in the PyPi distribution.

Fetching from GitHub is easier, so I'll do that first. I can always migrate back to `fetchPypi` once the tests are distributed.

I just hope their tests can run in a sandbox. :crossed_fingers:

Reply:
I now added tests, but there were a few complications:
1. A couple of tests required special devices and one required port redirection (I disabled and commented on them).
2. Tests take a very long time. I added a `pytest` option to print out how long each individual test takes. I'm at an impasse here and would like to ask for your advice on this. Should I disable some tests? If so, which ones? Could you please take a look at the build output and look at the durations?